core/src/global/ionic-global.ts

```diff
@@ -49,11 +49,17 @@ export default () => {
     config.set('animated', false);
   }
 
+  const isIonicElement = (elm: any) =>
+        elm.tagName && elm.tagName.startsWith('ION-');
+
+  const isAllowedIonicModeValue = (elmMode: string) =>
+      ['ios', 'md'].includes(elmMode);
+
   setMode((elm: any) => {
     while (elm) {
       const elmMode = (elm as any).mode || elm.getAttribute('mode');
 
-      if (elmMode) {
+      if (elmMode && (isIonicElement(elm) || isAllowedIonicModeValue(elmMode))) {
```

Comment:
Discussed with the FW team. They believe this would really be useful if it did something like the following completely untested code:

```typescript
  setMode((elm: any) => {
    while (elm) {
      const elmMode = (elm as any).mode || elm.getAttribute('mode');
      if (elmMode) {
        if (isAllowedIonicModeValue(elmMode)) {
          return elmMode;
        } else if (isIonicElement(elmMode)) {
          console.warn('Invalid ionic mode:', elmMode, 'expected: ios or md');
        }
      }
      elm = elm.parentElement;
    }
    return defaultMode;
  });
};
```

That way the user would be warned about the invalid mode on an Ionic element and the code would keep going up the tree rather than failing on that.

Would you be willing to make such a change?

Reply:
I did this. 🙂